### PR TITLE
Initial commit: Send Capabilities via mqtt

### PR DESCRIPTION
**Pull request by saschadoemer**

### CODE REVIEW

File: src/Service/Common/AbstractMessagingService.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<?php
+
```
Comment (saschadoemer) [inline src/Service/Common/AbstractMessagingService.php:2]: Remove empty lines.

File: src/Service/Common/AbstractMessagingService.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+namespace App\Service\Common;
+
+
```
Comment (saschadoemer) [inline src/Service/Common/AbstractMessagingService.php:6]: Remove empty lines.

File: tests/Helper/LoggerBuilder.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    use DateTimeZone;
+    use Monolog\Formatter\LineFormatter;
+    use Monolog\Handler\StreamHandler;
+    use Monolog\Logger;
+
+    /**
+     * The LoggerBuilder builds default logger for the tests.
+     * @package Lib\Tests\Helper
+     */
+    class LoggerBuilder
```
Comment (saschadoemer) [inline tests/Helper/LoggerBuilder.php:14]: This is not a builder pattern. Either create a real builder or rename the class to a `LoggerFactory`.

File: tests/Helper/PhpMqttClientBuilder.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
+<?php declare(strict_types=1);
+
```
Comment (saschadoemer) [inline tests/Helper/PhpMqttClientBuilder.php:2]: Remove empty lines.

File: tests/Helper/PhpMqttClientBuilder.php [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        private MqttClient $mqttClient;
+
+        /**
+         * Constructor.
+         * @param string $host
+         * @param string $port
+         * @param string $clientId
+         * @throws ProtocolNotSupportedException
+         */
+        public function __construct(string $host, string $port, string $clientId)
```
Comment (saschadoemer) [inline tests/Helper/PhpMqttClientBuilder.php:23]: Not a builder pattern >>> move parameters from the constructor to dedicated fluent methods.

File: tests/Service/Messaging/Mqtt/CapabilitiesServiceTest.php [hunk 6]
```diff
@@ -0,0 +1,75 @@
+<?php
```
Comment (saschadoemer) [inline tests/Service/Messaging/Mqtt/CapabilitiesServiceTest.php:1]: Since there is a new package for MQTT tests, former HTTP based tests should be moved to a dedicated package.

File: tests/Service/Messaging/Mqtt/CapabilitiesServiceTest.php [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    use function PHPUnit\Framework\assertNotNull;
+
+    class CapabilitiesServiceTest extends TestCase
+    {
+        /**
+         * @covers CapabilityService::send()
+         * @throws ProtocolNotSupportedException
+         * @throws Exception
+         */
+        public function testConnectMqttClientToFarmingSoftwareEndpoint()
```
Comment (saschadoemer) [inline tests/Service/Messaging/Mqtt/CapabilitiesServiceTest.php:29]: The name of the method does not match the test case.

File: tests/Service/Messaging/Mqtt/CapabilitiesServiceTest.php [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    use Lib\Tests\Applications\FarmingSoftware;
+    use Lib\Tests\Helper\Identifier;
+    use Lib\Tests\Helper\LoggerBuilder;
+    use Lib\Tests\Helper\OnboardResponseRepository;
+    use Lib\Tests\Helper\PhpMqttClientBuilder;
+    use PhpMqtt\Client\Exceptions\ProtocolNotSupportedException;
+    use PHPUnit\Framework\TestCase;
+    use function PHPUnit\Framework\assertNotNull;
+
+    class CapabilitiesServiceTest extends TestCase
```
Comment (saschadoemer) [inline tests/Service/Messaging/Mqtt/CapabilitiesServiceTest.php:22]: The test is missing a positive case and is only testing the negative outcome.

**Review by saschadoemer — COMMENTED** [review]
Please see comments.

File: tests/Service/Messaging/Mqtt/CapabilitiesServiceTest.php (hunk 6 above)
Comment (anschnit) [reply tests/Service/Messaging/Mqtt/CapabilitiesServiceTest.php:1]: I move the tests step by step as the mqtt implementation is ready. Is that ok?

Comment (saschadoemer) [reply tests/Service/Messaging/Mqtt/CapabilitiesServiceTest.php:1]: Fine for me.
